src/server/models/Game.js

```diff
@@ -63,55 +68,102 @@ export default class Game extends Model {
     return { ...this.createDirty(true), hand: player.hand.toIDArray() };
   }
 
-  async start() {
-    [
+  getKingdomCards() {
+    let a = [];
+    while (a.length < 10) {
+      let unusedCards = [];
+      Card.classes.forEach( c => {
+        if (c.supplyCategory === 'kingdom' && a.indexOf(c.title) === -1) unusedCards.push(c.title);
+      });
+      a.push(unusedCards[Math.floor(Math.random() * unusedCards.length)]);
+    }
+    if (0) {
+      return a;
+    } else {
+      return [
+      'Chapel',
+      'Cellar',
+      'Militia',
+      'Moat',
+      'Village'
+      ];
+    }
+  }
+
+  getSupplyCards() {
+    let SuppliesArray = [
       'Curse',
       'Copper',
       'Silver',
       'Gold',
       'Estate',
       'Duchy',
-      'Province',
-      'Bureaucrat',
-      'Cellar',
-      'Chapel',
-      'CouncilRoom',
-      'Festival',
-      'Gardens',
-      'Laboratory',
-      'Library',
-      'Market',
-      'Militia',
-      'Mine',
-      'Moat',
-      'Moneylender',
-      'Remodel',
-      'Smithy',
-      'ThroneRoom',
-      'Village',
-      'Witch',
-      'Workshop',
-      'Courtyard',
-      'Lurker',
-      'Bandit',
-      'Artisan',
-      'Harbinger',
-      'Merchant',
-      'Poacher',
-      'Sentry',
-      'Baron',
-      'Conspirator',
-      'Courtier',
-      'Diplomat',
-      'Ironworks',
-      'Mill',
-      'MiningVillage',
-      'Minion',
-      'Patrol',
-      'Swindler',
-      'WishingWell',
-
-    ].forEach((title) => {
+      'Province'];
+
+    const Kingdom = this.getKingdomCards();
+
+    this.log('Kingdom Contains:')
+    for (let i = 0; i < Kingdom.length; i++) {
+      this.log(`${Kingdom[i]}`);
+    }
+
+    if (Kingdom.indexOf('YoungWitch') !== -1) {
+      const possibleBanes = [];
+      Card.classes.forEach(c => {
+        if ((c.cost === 2 || c.cost === 3 ) &&
+        Kingdom.indexOf(c.title) === -1 && c.supplyCategory === 'kingdom') possibleBanes.push(c.title);
+      });
+      let Bane = possibleBanes[Math.floor(Math.random() * possibleBanes.length)];
+      Kingdom.push(Bane)
+      Card.classes.get(Bane).bane = true;
+      this.log(`${Bane} is Young Witch's Bane`);
+    }
+
+    Kingdom.forEach( title => {
+      Card.classes.get(title).dependancies.forEach( d => {
+        if (SuppliesArray.indexOf(d) === -1) {
+          SuppliesArray.push(d);
+        }
+      });
+    });
+
+    Kingdom.forEach(c => SuppliesArray.push(c));
+
+    return SuppliesArray;
+  }
+
+  endOfGame() {
+    let scores = [];
+    this.players.forEach(player => {
+      player.endOfGameCleanUp();
+      player.deck.forEach(c => {
+        player.vpTokens += c.getVpValue(player);
```

Comment:
The idea is that vpTokens will be incremented during the game by monument/goons/bishop/landmarks that give vp tokens/ farmers market etc.
Then at the end of the game we add our vp tokens to our points from our cards.

I've changed it to use a separate variable so that player.vpTokens is preserved and we can better display end of game stats at a future time, rather than just raw score.